base/libgit2/tag.jl

```diff
@@ -31,10 +31,13 @@ function tag_create(repo::GitRepo, tag::AbstractString, commit::Union{AbstractSt
 end
 
 function name(tag::GitTag)
-    return String(ccall((:git_tag_name, :libgit2), Cstring, (Ptr{Void}, ), tag.ptr))
+    str_ptr = ccall((:git_tag_name, :libgit2), Cstring, (Ptr{Void}, ), tag.ptr)
+    str_ptr == C_NULL && throw(Error.GitError(Error.ERROR))
+    return String(str_ptr)
 end
 
 function target(tag::GitTag)
-    oid_ptr = Ref(ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr))
-    return oid_ptr[]
+    oid_ptr = ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr)
+    oid_ptr == C_NULL && throw(Error.GitError(Error.ERROR))
+    return Oid(oid_ptr)
```

Comment:
Oh in this case the `Oid` constructor actually does check and throw for null pointers already.

Just so I'm clear, the old code was returning a `Ptr{Oid}` from the `ccall`, wrapping that in a `Ref` then dereferencing the `Ref` - what type was that returning, a ` Ptr{Oid}` or an `Oid`?